enhancements/cluster-api/installing-cluster-api-components-in-ocp.md

```diff
@@ -196,19 +196,20 @@ N/A
-
-If instead we think this is a hard requirement we will need to discuss and take measures to make this an option.
-
+In the current and in the implementation of the CAPI components installation, CVO installs the RBACs and CRDs for all the CAPI providers that are supported by OpenShift, instead of installing only the one that is directly compatible with the platform the cluster is running on.
+We consider good practice to be able to only install the RBACs/CRDs for the providers that we are going to effectively run in the cluster.
+As such here are some of the potential options being discussed:
+- a. leverage the same "transport ConfigMaps" mechanism we use for the other components (Deployments, Webhooks, Services, ...) of a CAPI provider and store the RBACs/CRDs in a second `<providername>-components` ConfigMap, which will be annotated, gathered and applied in a similar fashion to the already defined `<providername>` transport ConfigMap.
 
 #### Handling of multiple/custom CAPI providers
 
 If installing multiple infrastructure CAPI providers to be able to launch MachineSets on a different cloud provider than the one set in the `Infrastructure` Object status, is something considered a necessity, a potential way to achive is by keeping a list of the desired CAPI providers.
 
 By default the CAPI installer controller loads and applies all the providers ConfigMaps (a.k.a. "transport Configmaps") it finds (by filtering on the `provider.cluster.x-k8s.io/version` annotation).
-In order to let the user filter and selectively include certain providers ConfigMaps, the user could tweak a "providers-list" ConfigMap which would instruct the controller to install/uninstall the content of the mentioned ConfigMap in the cluster.
-
-This would also let the user specify custom transport ConfigMaps allowing them to deploy modified or alternative providers.
+In order to let the user filter and selectively include certain providers ConfigMaps, we will need to give them the ability to specify this somehow. Some potential options being discussed are:
+- a. expose a "providers-list" ConfigMap which is read by the operator at runtime and would instruct its controller to selectively install/uninstall the content of the referenced transport ConfigMaps.
+- b. in the operator, at runtime, perform detection of certain CAPI-specific FeatureGates that we would create for each supported provider. Then install the content of the transport ConfigMaps only for the enabled providers' FeatureGates. To do this we will need to have a way to set these FeatureGates at install/upgrade time based on the detected platform.
+- c. in the operator, at runtime, only detect the current platform by looking at the Infrastructure object's status, and solely install the content of that provider's transport ConfigMaps.
```

Comment:
This option for now is all that we strictly require. I think in the future, having a specific CRD to control the operator's behaviour would be an option, rather than a configmap, and would be the "correct" way to go from an API perspective. But for now, I think we should make installing multiple providers a non-goal, it doesn't seem to be necessary for this project and ironing out the kinks is delaying our progress right?

Reply:
Yes it's quite a lot more involved to support multiple providers.
And yes I think we can design `c` in a way that is forward compatible with a future change where we will have a CRD to control this.
Will go with `c` and address this in the non-goals.

Reply:
Done.